project14-Whack-a-Penguin/GameScene.swift

```diff
@@ -9,12 +9,118 @@ import SpriteKit
 import GameplayKit
 
 class GameScene: SKScene {
+    var slots: [WhackSlot] = []
+    var gameScore: SKLabelNode!
+    var popupTime = 0.85
+    var numRounds = 0
+    var score = 0 {
+        didSet {
+            gameScore.text = "Score: \(score)"
+        }
+    }
         
     override func didMove(to view: SKView) {
+        let background = SKSpriteNode(imageNamed: "whackBackground")
+        background.position = CGPoint(x: 512, y: 384)
+        background.blendMode = .replace
+        background.zPosition = -1
+        addChild(background)
+        
+        gameScore = SKLabelNode(fontNamed: "Chalkduster")
+        gameScore.text = "Score: 0"
+        gameScore.position = CGPoint(x: 8, y: 8)
+        gameScore.horizontalAlignmentMode = .left
+        gameScore.fontSize = 48
+        addChild(gameScore)
+        
+        for i in 0..<5 { createSlot(at: CGPoint(x: 100 + (i * 170), y: 410))}
+        for i in 0..<4 { createSlot(at: CGPoint(x: 180 + (i * 170), y: 320))}
+        for i in 0..<5 { createSlot(at: CGPoint(x: 100 + (i * 170), y: 230))}
+        for i in 0..<4 { createSlot(at: CGPoint(x: 180 + (i * 170), y: 140))}
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            [weak self] in
+            guard let self = self else { return }
+            self.creatEnemy()
+        }
         
     }
     
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        guard let touch = touches.first else { return }
+        let location = touch.location(in: self)
+        let tappedNodes = nodes(at: location)
+        
+        for node in tappedNodes {
+            guard let whackSlot = node.parent?.parent as? WhackSlot else {
+                continue }
+            if !whackSlot.isVisible { continue }
+            if whackSlot.isHit { continue }
```

Comment:
Very minor, more of a personal choice: I would put all tests in a single `guard`.

Reply:
That makes sense!  =D